Update Edge Functions docs (#36690)
supabase/supabase: apps/docs/content/guides/functions/troubleshooting.mdx

subtitle: 'How to solve common problems and issues related to Edge Functions.'
---

> When developing Edge Functions, you can run intue various issue during development, deployment, and at runtime. Most problems fall under these categories:

Reviewer: ```suggestion
When developing Edge Functions, you can run into various issues during development, deployment, and at runtime. Most problems fall under these categories:
```

### Unable to Call Edge Function

For invocation or CORS issues:

1. **Review CORS configuration:** Check out the [CORS guide](https://supabase.com/docs/guides/functions/cors), and ensure you've properly configured CORS headers
2. **Check function logs:** Look for errors in the Functions > Logs section
3. **Verify authentication:** Confirm JWT tokens and permissions are correct

```tsx
// ✅ Proper CORS handling
Deno.serve(async (req) => {
  if (req.method === 'OPTIONS') {
    return new Response(null, {
      status: 200,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Methods': 'POST, GET, OPTIONS',
        'Access-Control-Allow-Headers': 'Content-Type, Authorization',
      },
    })
  }

  // Your function logic here
  return new Response('Success', {
    headers: { 'Access-Control-Allow-Origin': '*' },
  })
})
```

There are two debugging tools available: Invocations and Logs. Invocations shows the Request and Response for each execution, while Logs shows any platform events, including deployments and errors.

---

## Local Development Issues

### Issues Serving Functions Locally

When `supabase functions serve` fails:

> 1. **Use debug mode:** Run with the `-debug` flag for detailed output

Reviewer: ```suggestion
1. **Use debug mode:** Run with the `--debug` flag for detailed output
```

2. **Check port availability:** Ensure ports `54321` and `8081` are available

```bash
# Serve with debug output
supabase functions serve your-function --debug

# Check specific port usage
lsof -i :54321
```

> If the problem persists, search the [Edge Runtime](https://github.com/supabase/edge-runtime) and [CLI](https://github.com/supabase/cli) repositories for similar error messages. Support engineers can then try to run the provided sample code locally and see if they can reproduce the issue.

Reviewer: ```suggestion
If the problem persists, search the [Edge Runtime](https://github.com/supabase/edge-runtime) and [CLI](https://github.com/supabase/cli) repositories for similar error messages.
```
